app/helpers/dossier_helper.rb

```diff
@@ -79,7 +79,11 @@ def class_badge_state(state)
 
   def status_badge(state, alignment_class = '')
     status_text = dossier_display_state(state, lower: true)
-    tag.span(status_text, class: "fr-badge fr-badge--sm #{class_badge_state(state)} fr-badge--no-icon #{alignment_class}", role: 'status')
+    if [Dossier.states.fetch(:en_instruction), Dossier.states.fetch(:accepte)].include?(state)
+      tag.span(status_text, class: "fr-badge fr-badge--sm #{class_badge_state(state)} #{alignment_class}", role: 'status')
+    else
+      tag.span(status_text, class: "fr-badge fr-badge--sm #{class_badge_state(state)} fr-badge--no-icon #{alignment_class}", role: 'status')
+    end
```

Comment:
je propose un refacto que je pense + lisible et évolutif, mais je te laisse aviser : 
 
```suggestion
      tag.span status_text, role: 'status', class: class_names(
        'fr-badge fr-badge--sm' => true,
        'fr-badge--no-icon' => [Dossier.states.fetch(:en_instruction), Dossier.states.fetch(:accepte)].include?(state),
        class_badge_state(state) => true,
        alignment_class => true
      )
```